apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/preadj/McsClaimTransformerV2.java

```diff
@@ -61,78 +84,189 @@ private static Claim transformClaim(PreAdjMcsClaim claimGroup) {
+                    .setValue(claimGroup.getIdrClaimMbi())))
+        .setName(getBeneName(claimGroup))
+        .setGender(
+            claimGroup.getIdrBeneSex() == null
+                ? null
+                : GENDER_MAP.get(claimGroup.getIdrBeneSex().toLowerCase()))
+        .setId("patient");
   }
 
-  private static List<Identifier> getIdentifier(PreAdjMcsClaim claimGroup) {
-    return Collections.singletonList(
-        new Identifier()
-            .setType(
-                new CodeableConcept(
+  private static List<HumanName> getBeneName(PreAdjMcsClaim claimGroup) {
+    HumanName name =
+        new HumanName()
+            .setFamily(
+                claimGroup.getIdrBeneLast_1_6() == null || claimGroup.getIdrBeneLast_1_6().isEmpty()
+                    ? null
+                    : claimGroup.getIdrBeneLast_1_6().charAt(0) + ".");
+
+    if (claimGroup.getIdrBeneFirstInit() != null || claimGroup.getIdrBeneMidInit() != null) {
+      name.setGiven(
+          List.of(
+              new StringType(
+                  claimGroup.getIdrBeneFirstInit() == null
+                      ? null
+                      : claimGroup.getIdrBeneFirstInit() + "."),
+              new StringType(
+                  claimGroup.getIdrBeneMidInit() == null
+                      ? null
+                      : claimGroup.getIdrBeneMidInit() + ".")));
+    }
+
+    return List.of(name);
+  }
+
+  private static Resource getContainedProvider(PreAdjMcsClaim claimGroup) {
+    Organization organization = new Organization();
+
+    if (claimGroup.getIdrBillProvType() != null) {
+      organization
+          .getExtension()
+          .add(
+              new Extension(BBCodingSystems.MCS_IDR_BILL_PROV_TYPE)
+                  .setValue(
+                      new Coding(
+                          BBCodingSystems.MCS_IDR_BILL_PROV_TYPE,
+                          claimGroup.getIdrBillProvType(),
+                          null)));
+    }
+
+    if (claimGroup.getIdrBillProvSpec() != null) {
+      organization
+          .getExtension()
+          .add(
+              new Extension(BBCodingSystems.PROVIDER_SPECIALTY)
+                  .setValue(
+                      new Coding(
+                          BBCodingSystems.PROVIDER_SPECIALTY,
+                          claimGroup.getIdrBillProvSpec(),
+                          null)));
+    }
+
+    if (claimGroup.getIdrBillProvEin() != null) {
+      organization
+          .getIdentifier()
+          .add(
+              new Identifier()
+                  .setType(
+                      new CodeableConcept(
+                          new Coding(
+                              C4BBOrganizationIdentifierType.TAX.getSystem(),
+                              C4BBOrganizationIdentifierType.TAX.toCode(),
+                              C4BBOrganizationIdentifierType.TAX.getDisplay())))
+                  .setSystem(BBCodingSystems.TAX_NUM)
```

Comment:
We'll need to add the system https://bluebutton.cms.gov/resources/variables/mcs/idr-bill-prov-ein for this value